presto-main/src/main/java/com/facebook/presto/execution/NodeTaskMap.java

```diff
@@ -108,87 +130,154 @@ private void addTask(RemoteTask task)
+                if (value < 0) {
+                    long oldValue = this.value.getAndSet(0L);
+                    totalValue.addAndGet(-oldValue);
+                    throw new IllegalArgumentException(stat + " is negative");
                 }
 
-                int oldValue = localPartitionedSplitCount.getAndSet(partitionedSplitCount);
-                nodeTotalPartitionedSplitCount.addAndGet(partitionedSplitCount - oldValue);
+                long oldValue = this.value.getAndSet(value);
+                totalValue.addAndGet(value - oldValue);
             }
 
             public void cleanup()
             {
-                int leakedSplits = localPartitionedSplitCount.getAndSet(0);
-                if (leakedSplits == 0) {
+                long leakedValues = value.getAndSet(0);
+                if (leakedValues == 0) {
                     return;
                 }
 
-                log.error("BUG! %s for %s leaked with %s partitioned splits.  Cleaning up so server can continue to function.",
+                log.error("BUG! %s for %s leaked with %s %s.  Cleaning up so server can continue to function.",
                         getClass().getName(),
                         taskId,
-                        leakedSplits);
+                        leakedValues,
+                        stat);
 
-                nodeTotalPartitionedSplitCount.addAndGet(-leakedSplits);
+                totalValue.addAndGet(-leakedValues);
             }
 
             @Override
             public String toString()
             {
                 return toStringHelper(this)
                         .add("taskId", taskId)
-                        .add("splits", localPartitionedSplitCount)
+                        .add(stat, value)
                         .toString();
             }
         }
+
+        // tracks stats which are passed as accumulated (cpu time) by calculating delta / duration.
+        @ThreadSafe
+        private class AccumulatedTaskStatsTracker
+                extends TaskStatsTracker
+        {
+            private final long windowSizeInMillis;
+            SortedMap<Long, Long> values = new TreeMap<>();
+
+            public AccumulatedTaskStatsTracker(String stat, TaskId taskId, AtomicLong totalValue, long windowSizeInMillis)
+            {
+                super(stat, taskId, totalValue);
+                this.windowSizeInMillis = windowSizeInMillis;
+            }
+
+            private long getDeltaPerSecond(long taskAgeInMillis, long value)
+            {
+                if (value > 0) {
+                    values.put(taskAgeInMillis, value);
+                    // this clears the map and make items eligible for GC
+                    values.headMap(values.lastKey() - windowSizeInMillis).clear();
+                    if (values.size() > 1) {
+                        long deltaValue = (values.get(values.lastKey()) - values.get(values.firstKey())) * 100;
```

Comment:
Why multiply by 100?

Reply:
This was there from the initial implementation. It was helping to calculate a percentage. I will remove.

Reply:
Actually, using percentage as utilization unit is a common practice. I will keep 100 and change the of the method to get CPU utilization.

Reply:
There's methods which don't require boxing: e.g. `lastLongKey`, `firstLongKey`